### ISSUE TITLE
shall convert simple urls to be clickable

### ISSUE THREAD

**Issue by henning** [stewart/gfm#2]
Hi,

The github markdown also includes the conversion of urls to clickable links. 
gfm seems not to do it, or im doing something wrong, do i?

Python 2.7.5+ (default, Feb 27 2014, 19:37:08) 
[GCC 4.8.1] on linux2
Type "help", "copyright", "credits" or "license" for more information.

> > > import gfm
> > > gfm.gfm('http://www.sprang.de')
> > > 'http://www.sprang.de'
> > > gfm.gfm(' http://www.sprang.de ')
> > > ' http://www.sprang.de '
> > > gfm.markdown(' http://www.sprang.de ')
> > > u'<p>http://www.sprang.de </p>'

**Comment by stewart** [comment]
You're right, this is missing functionality. I'll look into it a bit later today, thanks for letting me know!

**Comment by stewart** [comment]
Poked into this a bit, and I can't find or construct a good URL regex that won't also match existing Markdown-formatted URLs.

Going to leave this open for now, will look into it at some point in the future but for right now I don't have a quick fix.

Here's my idea of how the regex should match:

```
The link in this sentence (https://google.com/) should be matched.
Additionally, so should the one in this sentence - https://google.com/test?urlParams=true#hash.

However, this one shouldn't be - [https://github.com](https://github.com)
And neither should this one or it's expansion on the next line: [https://github.com](github).

[github]: https://github.com
```

**Comment by henning** [comment]
I found that py-gfm  https://github.com/dart-lang/py-gfm does all i need quite well... you might look at how they implement it...
